src/Symfony/Component/Validator/Constraints/EmailValidator.php

```diff
@@ -37,7 +37,7 @@ public function validate($value, Constraint $constraint)
         $valid = filter_var($value, FILTER_VALIDATE_EMAIL);
 
         if ($valid) {
-            $host = substr($value, strpos($value, '@') + 1);
+            $host = $this->extractHostname($value);
```

Comment:
We prefer to *not* test private methods. In this case, I agree with @xabbuh for inlining, and the test case should use the public interface only.